src/client/api/event.c

```diff
@@ -612,22 +612,31 @@ daos_eq_create(daos_handle_t *eqh)
 {
 	struct daos_eq_private	*eqx;
 	struct daos_eq		*eq;
+	crt_context_t		eq_ctx;
 	int			rc = 0;
 
 	/** not thread-safe, but best effort */
 	if (eq_ref == 0)
 		return -DER_UNINIT;
 
+	rc = crt_context_create(&eq_ctx);
+	if (rc != 0) {
+		D_ERROR("failed to create CART context: "DF_RC"\n", DP_RC(rc));
+		return rc;
+	}
```

Comment:
I would still fallback to the global context here instead of failing the eq creation for some pathological case where many EQ are being created.

Reply:
sure will fix that now.